proto.h

```diff
@@ -1213,26 +1213,31 @@ PERL_CALLCONV HV*	Perl_get_hv(pTHX_ const char *name, I32 flags);
 #define PERL_ARGS_ASSERT_GET_HV	\
 	assert(name)
 PERL_CALLCONV const char*	Perl_get_no_modify(pTHX)
+			__attribute__deprecated__
 			__attribute__warn_unused_result__
 			__attribute__pure__;
 #define PERL_ARGS_ASSERT_GET_NO_MODIFY
 
 PERL_CALLCONV char**	Perl_get_op_descs(pTHX)
+			__attribute__deprecated__
 			__attribute__warn_unused_result__
 			__attribute__pure__;
 #define PERL_ARGS_ASSERT_GET_OP_DESCS
 
 PERL_CALLCONV char**	Perl_get_op_names(pTHX)
+			__attribute__deprecated__
```

Comment:
We need to remove their use from `Opcode.xs` else we get:

```
gcc -c   -fwrapv -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wno-declaration-after-statement -Wc++-compat -Wwrite-strings -g   -DVERSION=\"1.57\" -DXS_VERSION=\"1.57\" -fPIC "-I../.."   Opcode.c
Opcode.xs: In function ‘op_names_init’:
Opcode.xs:59:5: warning: ‘Perl_get_op_names’ is deprecated [-Wdeprecated-declarations]
   59 |     op_names = get_op_names();
      |     ^~~~~~~~
In file included from ../../perl.h:5798,
                 from Opcode.xs:3:
../../proto.h:1227:22: note: declared here
 1227 | PERL_CALLCONV char** Perl_get_op_names(pTHX)
      |                      ^~~~~~~~~~~~~~~~~
Opcode.xs: In function ‘XS_Opcode_opset_to_ops’:
Opcode.xs:379:5: warning: ‘Perl_get_op_descs’ is deprecated [-Wdeprecated-declarations]
  379 |     char **names = (desc) ? get_op_descs() : get_op_names();
      |     ^~~~
In file included from ../../perl.h:5798,
                 from Opcode.xs:3:
../../proto.h:1221:22: note: declared here
 1221 | PERL_CALLCONV char** Perl_get_op_descs(pTHX)
      |                      ^~~~~~~~~~~~~~~~~
Opcode.xs:379:5: warning: ‘Perl_get_op_names’ is deprecated [-Wdeprecated-declarations]
  379 |     char **names = (desc) ? get_op_descs() : get_op_names();
      |     ^~~~
In file included from ../../perl.h:5798,
                 from Opcode.xs:3:
../../proto.h:1227:22: note: declared here
 1227 | PERL_CALLCONV char** Perl_get_op_names(pTHX)
      |                      ^~~~~~~~~~~~~~~~~
Opcode.xs: In function ‘XS_Opcode_opdesc’:
Opcode.xs:470:5: warning: ‘Perl_get_op_descs’ is deprecated [-Wdeprecated-declarations]
  470 |     char **op_desc = get_op_descs();
      |     ^~~~
In file included from ../../perl.h:5798,
                 from Opcode.xs:3:
../../proto.h:1221:22: note: declared here
 1221 | PERL_CALLCONV char** Perl_get_op_descs(pTHX)
      |                      ^~~~~~~~~~~~~~~~~
```

Hence #19785, which does this then your commit here.

I propose we close this PR and go with that one (or a variant of it)